modDbImpl/src/org/aion/db/impl/leveldb/LevelDB.java

```diff
@@ -199,312 +175,187 @@ public void close() {
         } finally {
             // ensuring the db is null after close was called
             db = null;
-            // releasing write lock
-            lock.writeLock().unlock();
         }
     }
 
     @Override
     public void compact() {
-        // acquire read lock
-        lock.writeLock().lock();
-
         LOG.info("Compacting " + this.toString() + ".");
```

Comment:
Minor point, not sure if this messages needs to be logged to info, could become noisy